### PR TITLE
Adds randomly generated sea slugs.

### PR DESCRIPTION
**Pull request by Zuhayr**
Sluuuugs

### CODE REVIEW

File: code/modules/europa/mobs/living/slugs.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	for(var/component in components)
+		cache_key += "\[[component]-[base_colour[component]][rider[component]]\]"
+	if(!slug_cache[cache_key])
+		var/icon/blank = icon(icon,"")
+		for(var/component in components)
+			var/slug_key = "[component]-[base_colour[component]][rider[component]]"
+			if(!slug_cache[slug_key])
+				var/icon/base = icon(icon, "[component][rider[component]]")
+				base.Blend(base_colour[component],ICON_ADD)
+				blank.Blend(base, ICON_OVERLAY)
```
Comment (GinjaNinja32) [inline code/modules/europa/mobs/living/slugs.dm]: Convert the icons to pale, use multiplicative blending via `image.color` and images' overlay list and you can avoid these icon operations.

Comment (Zuhayr) [inline code/modules/europa/mobs/living/slugs.dm]: Colour var looks a lot worse than this imo.

**Comment by Finalsong** [comment]
Man, I hope these guys make a re-appearance. Seaslugs are great.

**Comment by Zuhayr** [comment]
Fuuuuuck.

**Pull request closed without merging**